server/src/test/java/com/paypal/selion/grid/servlets/BaseGridRegistyServletTest.java

```diff
@@ -0,0 +1,92 @@
+|                                                                                                                     |
+|  Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance     |
+|  with the License.                                                                                                  |
+|                                                                                                                     |
+|  You may obtain a copy of the License at                                                                            |
+|                                                                                                                     |
+|       http://www.apache.org/licenses/LICENSE-2.0                                                                    |
+|                                                                                                                     |
+|  Unless required by applicable law or agreed to in writing, software distributed under the License is distributed   |
+|  on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for  |
+|  the specific language governing permissions and limitations under the License.                                     |
+\*-------------------------------------------------------------------------------------------------------------------*/
+
+package com.paypal.selion.grid.servlets;
+
+import static org.testng.Assert.assertEquals;
+import static org.testng.Assert.assertTrue;
+
+import org.openqa.grid.common.RegistrationRequest;
+import org.openqa.grid.internal.Registry;
+import org.openqa.grid.internal.RemoteProxy;
+import org.openqa.grid.internal.utils.DefaultCapabilityMatcher;
+import org.openqa.grid.internal.utils.GridHubConfiguration;
+import org.openqa.grid.selenium.proxy.DefaultRemoteProxy;
+import org.openqa.grid.web.Hub;
+import org.springframework.mock.web.MockHttpServletResponse;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import javax.servlet.http.HttpServletResponse;
+
+public class BaseGridRegistyServletTest {
+    protected static Registry registry;
+    protected static Hub hub;
+
+    public void beforeClass() {
+        // Create a Selenium grid hub configuration
+        GridHubConfiguration hubConfig = new GridHubConfiguration();
+        hubConfig.setPort(4321);
+        hubConfig.setCapabilityMatcher(new DefaultCapabilityMatcher());
+
+        // Create a Hub instance
+        hub = new Hub(hubConfig);
+
+        // Create a Selenium grid registry, using the new hubConfig
+        registry = Registry.newInstance(hub, hubConfig);
+
+        // Create a Selenium grid registration request
+        Map<String, Object> configuration = new HashMap<>();
+        configuration.put(RegistrationRequest.REMOTE_HOST, "http://127.0.0.1:1234");
+        configuration.put(RegistrationRequest.ID, "127.0.0.1");
```

Comment:
![Codacy](https://www.codacy.com/versioned/images/favicon.png) Issue found: [Do not hard code the IP address](https://www.codacy.com/app/SeLion/paypal-SeLion/file/2093723322/issues/source?bid=2928425&fileBranchId=3141724#l54)